src/components/external/CapitalOverview/components/GrantAction/GrantActionContainer.tsx

```diff
@@ -0,0 +1,176 @@
+import { FunctionalComponent } from 'preact';
+import { useCallback, useMemo, useState } from 'preact/hooks';
+import useCoreContext from '../../../../../core/Context/useCoreContext';
+import useTimezoneAwareDateFormatting from '../../../../../hooks/useTimezoneAwareDateFormatting';
+import { DATE_FORMAT_CAPITAL_OVERVIEW, DATE_FORMAT_MISSING_ACTION } from '../../../../../constants';
+import { GRANT_ACTION_CLASS_NAMES } from './constants';
+import './GrantAction.scss';
+import Alert from '../../../../internal/Alert/Alert';
+import { AlertTypeOption } from '../../../../internal/Alert/types';
+import Button from '../../../../internal/Button';
+import { useConfigContext } from '../../../../../core/ConfigContext';
+import { EMPTY_OBJECT } from '../../../../../utils';
+import { getTopWindowHref, setTopWindowHref } from './utils';
+import { ButtonVariant } from '../../../../internal/Button/types';
+import { IGrant } from '../../../../../types';
+import useMutation from '../../../../../hooks/useMutation/useMutation';
+
+type ActionType = NonNullable<IGrant['missingActions']>[number]['type'];
+
+export const GrantActionContainer: FunctionalComponent<{ missingActions: IGrant['missingActions']; offerExpiresAt?: string; className?: string }> = ({
+    missingActions = [],
+    offerExpiresAt,
+    className,
+}) => {
+    const { i18n, updateCore } = useCoreContext();
+    const { dateFormat } = useTimezoneAwareDateFormatting();
+    const { endpoints } = useConfigContext();
+
+    const ACTION_CONFIG = useMemo(
+        () =>
+            ({
+                signToS: {
+                    getTitle: (formattedDate: string | undefined) =>
+                        `${i18n.get('capital.signTermsAndConditionsToReceiveFunds')}${
+                            formattedDate ? ` ${i18n.get('capital.thisOfferExpiresOn', { values: { date: formattedDate } })}` : ''
+                        }`,
+                    buttonLabelKey: 'capital.goToTermsAndConditions',
+                },
+                AnaCredit: {
+                    getTitle: (formattedDate: string | undefined) =>
+                        i18n.get(
+                            formattedDate
+                                ? 'capital.weNeedABitMoreInformationToProcessYourFundsPleaseCompleteThisActionBy'
+                                : 'capital.weNeedABitMoreInformationToProcessYourFundsPleaseCompleteThisAction',
+                            { values: formattedDate ? { date: formattedDate } : undefined }
+                        ),
```

Comment:
This should be a bit cleaner, as it avoids passing params to the copy which does not accept them.
```suggestion
                        formattedDate
                          ?  i18n.get('capital.weNeedABitMoreInformationToProcessYourFundsPleaseCompleteThisActionBy', { values: { date: formattedDate } }
                          : i18n.get('capital.weNeedABitMoreInformationToProcessYourFundsPleaseCompleteThisAction'),
```

Reply:
✅